### ISSUE TITLE
"the" in a Matrix room comes out as "@undefined" on the Slack side

### ISSUE THREAD

**Issue by cmchittom** [matrix-org/matrix-appservice-slack#166]
Sometimes the word "the" on the Matrix side is replaced by "@undefined" in the Slack channel.

There is a Slack user whose display name starts with "the". In very limited testing, the issue did occur in a channel that user is in, but was not able to reproduce in a channel which that user is not in.

If it matters, this is Synapse 0.99.5.2 running on FreeBSD 12.  Please let me know if you need any additional information.

**Comment by cmchittom** [comment]
I see this has been merged. Pardon my ignorance, but do I need to do anything in particular to take advantage of this fix? I added this integration via the Riot UI, rather than as something standalone.

**Comment by Cadair** [comment]
If @Half-Shot has updated the matrix.org deployment then you wont have to do anything.

**Comment by cmchittom** [comment]
I'm not sure whether @Half-Shot has updated it or not, but I thought I would note that this problem is still occurring.  It also occurs with the word "a"--even the letter a inside some words (but weirdly, not every time).

Possibly related, the bridge matches also "will" (i.e., just that bare word, not "@will") to a Slack user with the displayname will, thus making him think he's being mentioned when he's not.

**Comment by will** [comment]
yeah it fuckin sucks having your name as "will" on slack cause they wont make the option to notify without the @ a user setting, it's admin only and they don’t give a fuck and wont fix it no matter how many times you report it to them because slack is a dumb fucking company

it also sucks on github sometimes like now

**Comment by cmchittom** [comment]
Blarg, sorry about that.
